### PR TITLE
Fix test failures from internal presubmit

### PR DESCRIPTION
**Pull request by wu-hui**
1. Use proper executor service for spec tests.
2. Skip `@Hide` annotated Task interface.

### CODE REVIEW

File: firebase-firestore/src/test/java/com/google/firebase/firestore/LoadBundleTaskTest.java [hunk 1]
```diff
@@ -61,7 +61,20 @@
 
   @Test
   public void testImplementsAllTaskInterface() {
+    // Check if the internal gms Hide annotation is accessible.
+    Class hideClazz = null;
+    try {
+      hideClazz = Class.forName("com.google.android.gms.common.internal.Hide");
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/LoadBundleTaskTest.java:67]: If we found this class, should we just exit the test?

**Review by schmidt-sebastian — APPROVED** [review]

File: firebase-firestore/src/test/java/com/google/firebase/firestore/LoadBundleTaskTest.java (hunk 1 above)
Comment (wu-hui) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/LoadBundleTaskTest.java:67]: Why? The presence of this class will only indicate we are running on internal platform, we still need to implement all methods, isn't it?

**Review by wu-hui — COMMENTED** [review]
Merging it for now. We can apply your suggestion later if that is what is needed here.

File: firebase-firestore/src/test/java/com/google/firebase/firestore/LoadBundleTaskTest.java (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/LoadBundleTaskTest.java:67]: Please disregard. I though the statements in line 74 were or-ed together. LGTM.

**Pull request merged**